node/src/components/consensus/highway_core/active_validator.rs

```diff
@@ -224,9 +221,14 @@ impl<C: Context> ActiveValidator<C> {
         timestamp: Timestamp,
         state: &State<C>,
     ) -> bool {
+        let earliest_vote_time = self.earliest_vote_time(state);
+        if timestamp < earliest_vote_time {
+            error!(%earliest_vote_time, %timestamp, "earliest_vote_time is greater than current time stamp");
```

Comment:
Maybe a warning is enough for this? `timestamp < earliest_vote_time` can happen due to network issues, sender delays etc.

Reply:
I don't think it can happen directly due to network issues. `earliest_vote_time` is the timestamp of our own latest message, so this can only happen if a timer event got delayed so much that in the meantime we've already created a vertex with a later timestamp.

Reply:
But `timestamp` is a timestamp from the incoming message, right?

Reply:
No, it used to be that, as a workaround. But with this PR, it will be the current timestamp, and thus the timestamp we would use for our confirmation vote.

Reply:
(We access `vote.timestamp` below.)

Reply:
Thanks. I got lost a bit. How about renaming `timestamp` argument to `now`? Looks like this pattern is employed in other places.

Reply:
I remember we talked about that last week. I forgot why we ended up with `timestamp` again. Sounds like it should be `now`, yes.